src/btree/bt_curnext.c

```diff
@@ -921,7 +921,7 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
                 WT_ERR(__wt_page_dirty_and_evict_soon(session, cbt->ref));
                 WT_STAT_CONN_INCR(session, cache_eviction_force_obsolete_delete);
             } else if (page->type != WT_PAGE_COL_FIX && total_skipped != 0 &&
-              !cbt->non_deleted_updates) {
+              !cbt->non_deleted_updates && session->txn->isolation != WT_ISO_READ_UNCOMMITTED) {
```

Comment:
I think for hs, I don't know whether we want to evict or not evict in this case? This check excludes the hs table as it is always read/written with read uncommitted isolation. If we choose to evict hs pages here, they will always be evicted after the walk which is also not what we want.

Reply:
We will not skip any of the history store pages as part of the tree walk. So evicting the history store pages will not provide any benefit. So ignoring HS is fine.

```
        if (!F_ISSET(&cbt->iface, WT_CURSTD_KEY_ONLY) &&
          session->txn->isolation == WT_ISO_SNAPSHOT &&
          !F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE))
            WT_ERR(
              __wt_tree_walk_custom_skip(session, &cbt->ref, __wt_btcur_skip_page, NULL, flags));
```

Reply:
Yeah, we ignore the tombstone for the history store. Can we add this to the comment?

Reply:
Added more comments. Please check now.